### DISCUSSION TITLE
In Mirth How can i get RECEIVED Messages through API messages/count

### DISCUSSION THREAD

**Discussion by BhargaviRM** [nextgenhealthcare/connect#6018] [Q&A] [answered]
Hi how to get Received messages, in messages/count API it showing received messages as 0 but in mirth dashboard received messages are there, how to get received messages count through API

received message count for all channels is consistently showing as zero, while counts for other statuses are being retrieved correctly.

**Comment by pacmano1** [comment]
Post a copy of the API call you made.     And received is likely to be 0 once your messages process.

Do the same search in the message browser and select "recieved" at the top and then type "search".    

You are searching on this in the UI:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/74b03e81-2944-4550-a964-76c6059d3c4b)

**Reply by pacmano1** [reply]
read my post again.

**Reply by BhargaviRM** [reply]
![image](https://github.com/nextgenhealthcare/connect/assets/106804698/1e292d59-58d3-4cfe-bbc9-dab14219e4e2)


here i am seeing

how can i count this received messages

**Reply by pacmano1** [reply]
**And you have a mirth instance on the internet accesible to anyone with the default user name and password.   And version 3.8.1 (which has critical vulnerabilites).**

You should never do that.

**Reply by BhargaviRM** [reply]
ok i will update

**Reply by pacmano1** [reply, marked as answer]
Those are stats, call https://3.108.67.239:8443/api/#!/Channel_Statistics/getStatistics

**Reply by pacmano1** [reply]
Changing the default password is not enough. There are other links but read: https://digital.nhs.uk/cyber-alerts/2023/cc-4400
